### PR TITLE
Add Local User (Add/Set/Remove)

### PR DESCRIPTION
**Pull request by CedricMoreau**

### CODE REVIEW

File: Tests/integration/UserLocal.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        It "Change email to" {
+            Get-FGTUserLocal -name $pester_userlocal | Set-FGTUserLocal -email_to "powerfgt@power.fgt"
+            $userlocal = Get-FGTUserLocal -name $pester_userlocal
+            $userlocal.name | Should -Be $pester_userlocal
+            $userlocal.status | Should -Be "disable"
+            $userlocal.'email-to' | Should -Be "powerfgt@power.fgt"
+            $userlocal.'two-factor' | Should -Be "disable"
+        }
+
+        It "Change MFA" {
```
Comment (alagoutte) [inline Tests/integration/UserLocal.Tests.ps1:141]: enable MFA ?

File: Tests/integration/UserLocal.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        It "Change Name" {
+            Get-FGTUserLocal -name $pester_userlocal | Set-FGTUserLocal -name "pester_userlocal_change"
+            $userlocal = Get-FGTUserLocal -name "pester_userlocal_change"
+            $userlocal.name | Should -Be "pester_userlocal_change"
+            $userlocal.status | Should -Be "disable"
+            $userlocal.'email-to' | Should -Be "powerfgt@power.fgt"
+            $userlocal.'two-factor' | Should -Be "email"
+        }
+
+        AfterAll {
```
Comment (alagoutte) [inline Tests/integration/UserLocal.Tests.ps1:159]: use AfterEach ?

**Review by alagoutte — CHANGES_REQUESTED** [review]

File: PowerFGT/Public/cmdb/user/local.ps1 [hunk 3]
```diff
@@ -5,6 +5,129 @@
...
+        [switch]$status,
+        [Parameter (Mandatory = $false, ParameterSetName = "local")]
+        [string]$password,
+        [Parameter (Mandatory = $false, ParameterSetName = "radius")]
+        [string]$radius_server,
+        [Parameter (Mandatory = $false, ParameterSetName = "tacacs")]
+        [string]$tacacs_server,
+        [Parameter (Mandatory = $false)]
+        [ValidateSet("fortitoken", "email", "sms", "disable", "fortitoken-cloud")]
+        [string]$two_factor,
```
[the remark this answers is not in the document]
Comment (CedricMoreau) [reply PowerFGT/Public/cmdb/user/local.ps1:41]: Why

[the remark this answers is not in the document]
Comment (alagoutte) [reply PowerFGT/Public/cmdb/user/local.ps1:41]: avoid duplicate entry...

File: PowerFGT/Public/cmdb/user/local.ps1 [hunk 4]
```diff
@@ -1,10 +1,144 @@
...
+        Add Local User object name FGT, password mypassword and enable it, with two factor authentication by fortitoken
+    #>
+
+    Param(
+        [Parameter (Mandatory = $true)]
+        [string]$name,
+        [Parameter (Mandatory = $false)]
+        [switch]$status,
+        [Parameter (Mandatory = $false, ParameterSetName = "local")]
+        [SecureString]$passwd,
```
Comment (alagoutte) [inline PowerFGT/Public/cmdb/user/local.ps1:41]: $password ?

File: Tests/integration/UserLocal.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            Get-FGTUserLocal
+        } | Should -Not -Throw
+    }
+
+    It "Get ALL User Local" {
+        $userlocal = Get-FGTUserLocal
+        $userlocal.count | Should -Not -Be $NULL
+    }
+
+    It "Get ALL User Local with -skip" {
```
Comment (alagoutte) [inline Tests/integration/UserLocal.Tests.ps1:32]: add also with with -meta parameter

File: Tests/integration/UserLocal.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+    AfterAll {
+        Get-FGTUserLocal -name $pester_userlocal | Remove-FGTUserLocal -confirm:$false
+    }
+
+}
+
+Describe "Add User Local" {
+
+    Context "Local User (Email, MFA, etc" {
```
Comment (alagoutte) [inline Tests/integration/UserLocal.Tests.ps1:65]: missing )

File: Tests/integration/UserLocal.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+        It "Add User Local $pester_userlocal MFA by email" {
+            Add-FGTUserLocal -Name $pester_userlocal -status -two_factor email -email_to "powerfgt@power.fgt" -passwd $pester_userlocalpassword
+            $userlocal = Get-FGTUserLocal -name $pester_userlocal
+            $userlocal.name | Should -Be $pester_userlocal
+            $userlocal.status | Should -Be "enable"
+            $userlocal.'email-to' | Should -Be "powerfgt@power.fgt"
+            $userlocal.'two-factor' | Should -Be "email"
+            }
+
```
Comment (alagoutte) [inline Tests/integration/UserLocal.Tests.ps1:97]: missing some case like radius, tacacs (and ldap) server ?

File: Tests/integration/UserLocal.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+Describe "Remove User Local" {
+
+    Context "local" {
+
+        BeforeEach {
+            Add-FGTUserLocal -Name $pester_userlocal -passwd $pester_userlocalpassword
+        }
+
+        It "Remove User Local $pester_userlocal by pipeline" {
```
Comment (alagoutte) [inline Tests/integration/UserLocal.Tests.ps1:168]: remove also by id ?

File: PowerFGT/Public/cmdb/user/local.ps1 [hunk 9]
```diff
@@ -1,10 +1,144 @@
...
+        [string]$two_factor_authentication,
+        [Parameter (Mandatory = $false)]
+        [string]$two_factor_notification,
+        [Parameter (Mandatory = $false)]
+        [string]$fortitoken,
+        [Parameter (Mandatory = $false)]
+        [string]$email_to,
+        [Parameter (Mandatory = $false)]
+        [string]$sms_server,
+        [Parameter(Mandatory = $false)]
```
Comment (alagoutte) [inline PowerFGT/Public/cmdb/user/local.ps1:60]: add also -data parameter like other add function ;)

**Pull request merged**